api/sip/client.py

```diff
@@ -230,15 +230,20 @@ def patron_information(self, *args, **kwargs):
             
     def connect(self):
         """Create a socket connection to a SIP server."""
-        try:
-            sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
-        except socket.error, msg:
-            self.log.warn("Error initializing socket: %s", msg[1])
-
-        try:
-            sock.connect((self.target_server, self.target_port))
-        except socket.error, msg:
-            self.log.warn("Error connecting to SIP server: %s", msg)
+        with self.socket_lock:
+            try:
```

Comment:
At this point it would be useful to check whether the socket was refreshed while we were waiting for `socket_lock`. Otherwise we will probably wastefully create a second socket. But I don't see how we can check that the socket was just refreshed _and is still known to be good_.